tools/src/bin/sha256tree-benching.rs

```diff
@@ -0,0 +1,111 @@
+use clvmr::chia_dialect::ChiaDialect;
+use clvmr::run_program::run_program;
+use std::fs::File;
+use std::io::Write;
+use std::time::Instant;
+
+use linreg::linear_regression_of;
+
+// bring in your existing code
+use clvmr::allocator::Allocator;
+
+fn time_per_byte_for_atom(a: &mut Allocator, output: &mut dyn Write) -> (f64, f64) {
+    let mut samples = Vec::<(f64, f64)>::new();
+    let dialect = ChiaDialect::new(0x0200); // enable shatree
+
+    let op_code = a.new_number(63.into()).unwrap();
+    let quote = a.new_number(1.into()).unwrap();
+    let mut atom_str = "ff".repeat(10_000);
+    let checkpoint = a.checkpoint();
+
+    for i in 0..100000 {
+        // make the atom longer as a function of i
+        atom_str.push_str(&((i % 89) + 10).to_string().repeat(32)); // just to mix it up
+        let atom = a.new_atom(&hex::decode(&atom_str).unwrap()).unwrap();
+        // let args = a.new_pair(atom, a.nil()).unwrap();
+        let args = a.new_pair(quote, atom).unwrap();
+        let call = a.new_pair(args, a.nil()).unwrap();
+        let call = a.new_pair(op_code, call).unwrap();
+        let start = Instant::now();
+        run_program(a, &dialect, call, a.nil(), 11000000000).unwrap();
+        let duration = start.elapsed();
+        let sample = (i as f64, duration.as_nanos() as f64);
+        writeln!(output, "{}\t{}", sample.0, sample.1).expect("failed to write");
+        samples.push(sample);
+
+        a.restore_checkpoint(&checkpoint);
+    }
+
+    linear_regression_of(&samples).expect("linreg failed")
+}
+
+fn time_per_cons_for_list(a: &mut Allocator, output: &mut dyn Write) -> (f64, f64) {
```

Comment:
I thought you would fold this into the `benchmark-clvm-costs.rs`. I think seeing the plots would be helpful as well, as a confirmation that it looks right, and that it's linear.

Reply:
Integrated now